feat: Use functional integration importants for JS SDKs (#9056)
getsentry/sentry-docs: docs/platforms/react-native/integrations/default.mdx

description: "Learn more about system integrations: Dedupe, FunctionToString, Breadcrumbs, LinkedErrors, and HttpContext (UserAgent), which are integrated into the standard library or the interpreter itself by default."
---

> The below system integrations are part of the standard library or the interpreter itself and are enabled by default. To understand what they do and how to disable them if they cause issues, read on.

Author: I copied this here as it was only used here, as far as I can tell @krystofwoldrich . Eventually we'll need to update the integrations for react-native as well, I left them as-is for now!

```javascript
{
  // Log calls to `console.log`, `console.debug`, etc
  console: boolean;

  // Log all click and keypress events
  // - When an object with `serializeAttribute` key is provided,
  //   Breadcrumbs integration will look for given attribute(s) in DOM elements,
  //   while generating the breadcrumb trails.
  //   Matched elements will be followed by their custom attributes,
  //   instead of their `id`s or `class` names.
  dom: boolean | { serializeAttribute: string | string[] };

  // Log HTTP requests done with the Fetch API
  fetch: boolean;

  // Log calls to `history.pushState` and friends
  history: boolean;

  // Log whenever we send an event to the server
  sentry: boolean;

  // Log HTTP requests done with the XHR API
  xhr: boolean;
}
```

<PlatformSection notSupported={["react-native"]}>

### GlobalHandlers

_Import name: `Sentry.Integrations.GlobalHandlers`_

This integration attaches global handlers to capture uncaught exceptions and unhandled rejections.

Available options:

```javascript
{
  onerror: boolean;
  onunhandledrejection: boolean;
}
```

</PlatformSection>

### LinkedErrors

_Import name: `Sentry.Integrations.LinkedErrors`_

> This integration allows you to configure linked errors. They'll be recursively read up to a specified limit, and a lookup will be performed by a specific key on the captured Error object. By default, the integration records up to five errors and the key used for recursion is `"cause"`.
> 

Reviewer: ```suggestion
This integration allows you to configure linked errors which are recursively read up to a specified limit. A lookup by a specific key on the captured Error object is then performed. By default, the integration records up to five errors and the key used for recursion is `"cause"`.

```

This integration also enhances captured error events with data from an [AggregateError](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AggregateError) where applicable, by recursively iterating over the `errors` property on Error objects.

Available options:

```javascript
{
  key: string; // default: "cause"
  limit: number; // default: 5
}
```

> Here is a code example of how this could be implemented:

Reviewer: ```suggestion
Here's a code example of how this could be implemented:
```

  include: {
    cookies: boolean  // default: true,
    data: boolean  // default: true,
    headers: boolean  // default: true,
    ip: boolean  // default: false,
    query_string: boolean  // default: true,
    url: boolean  // default: true,
    user: boolean | {
      id: boolean  // default: true,
      username: boolean  // default: true,
      email: boolean  // default: true,
    },
  },
  // Controls how the transaction will be reported. Options are 'path' (`/some/route`),
  // 'methodPath' (`GET /some/route`), and 'handler' (the name of the route handler
  // function, if available)
  transactionNamingScheme: string  // default: 'methodPath',
};
```

</PlatformSection>

## Modifying System Integrations

To disable system integrations, set `defaultIntegrations: false` when calling `init()`.

To override their settings, provide a new instance with your config to the `integrations` option. For example, to turn off browser capturing console calls:

<SignInNote />

```javascript
Sentry.init({
  dsn: "___PUBLIC_DSN___",

  integrations: [
    new Sentry.Integrations.Breadcrumbs({
      console: false,
    }),
  ],
});
```

### Removing an Integration

> This example removes the default-enabled integration for adding breadcrumbs to the event:

Reviewer: ```suggestion
This example removes the integration for adding breadcrumbs to the event, which is enabled by default:
```
